### ISSUE TITLE
avoid cputime() calls

### ISSUE THREAD

**Issue by malb** [fplll/fplll#435]
Paul Zimmermann reports:

> In short, you have to change line 52 of fplll/lll.h (in fplll-5.3.2) as follows:
>
> -  int start_time = cputime();
> +  int start_time = (verbose) ? cputime() : 0;
>
> The reason is that cputime() calls getrusage(), and calling getrusage() in a multi-thread environment is a disaster. I guess you can do the same in the other functions that call cputime.

That is, calls to `cputime()` have a negative effect on multi-threaded performance. 

I see these occurrences:

``` shell
fplll/bkz.cpp:33:      cputime_start(0)
fplll/bkz.cpp:376:             (cputime() - cputime_start) * 0.001);
fplll/bkz.cpp:459:             (cputime() - cputime_start) * 0.001);
fplll/bkz.cpp:511:             (cputime() - cputime_start) * 0.001);
fplll/bkz.cpp:562:  cputime_start = cputime();
fplll/bkz.cpp:587:    if ((flags & BKZ_MAX_TIME) && (cputime() - cputime_start) * 0.001 >= param.max_time)
fplll/bkz.cpp:668:    dump_gso(param.dump_gso_filename, true, "Output", -1, (cputime() - cputime_start) * 0.001);
fplll/bkz.cpp:683:       << std::setw(9) << std::setprecision(3) << (cputime() - cputime_start) * 0.001 << "s";
fplll/lll.cpp:52:  int start_time = cputime();
fplll/lll.cpp:89:             << " cputime=" << cputime() - start_time << endl;
fplll/hlll.cpp:36:  int start_time = cputime();
fplll/hlll.cpp:45:    cerr << "Discovering vector 1/" << m.get_d() << " cputime=" << cputime() - start_time << endl;
fplll/hlll.cpp:71:    cerr << "Discovering vector 2/" << m.get_d() << " cputime=" << cputime() - start_time << endl;
fplll/hlll.cpp:125:                 << " cputime=" << cputime() - start_time << endl;
```

**Comment by joerowell** [comment]
Sorry if it's a stupid question: why do we care about the CPU time in these functions? That's probably the best way to decide upon a viable solution (other than just using the ``verbose`` flag).

**Comment by malb** [comment]
We use it for statistics and in BKZ for early abort, i.e. you can say: "run it for X time" (not sure anybody is using this)

**Comment by joerowell** [comment]
Fixed in #438 .
